lib/rspec/core/example.rb

```diff
@@ -138,8 +138,10 @@ def duplicate_with(metadata_overrides={})
 
         # don't clone the example group because the new example
         # must belong to the same example group (not a clone).
+        #
+        # block is nil in new_metadata so we have to get it from metadata.
         Example.new(example_group, description.clone,
-                    new_metadata, new_metadata[:block])
+                    new_metadata, metadata[:block])
```

Comment:
AFAIK, blocks are immutable, so I'm not sure what cloning it does for us other than taking more memory and time.